R/read_slf.R

```diff
@@ -3,185 +3,212 @@
-  # With testing it seems to usually be faster if we do partnership
-  # filtering before recid filtering
-  if (!is.null(partnerships)) {
-    slfs_list <- purrr::map(
-      slfs_list,
-      ~ dplyr::filter(.x, .x$hscp2018 %in% partnerships)
-    )
-  }
+      if (!is.null(recids)) {
+        slf_table <- dplyr::filter(
+          slf_table,
+          .data$recid %in% recids
+        )
+      }
+      if (!is.null(partnerships)) {
+        slf_table <- dplyr::filter(
+          slf_table,
+          .data$hscp2018 %in% partnerships
+        )
+      }
+      if (remove_partnership_var) {
+        slf_table <- dplyr::select(slf_table, -"hscp2018")
+      }
+      if (remove_recid_var) {
+        slf_table <- dplyr::select(slf_table, -"recid")
+      }
+
+      return(slf_table)
+    }
+  )
 
-  # If a recid is specified filter now
-  if (!is.null(recids)) {
-    slfs_list <- purrr::map(
-      slfs_list,
-      ~ dplyr::filter(.x, .x$recid %in% recids)
-    )
+  # Collapse the list to a single Arrow Table object
+  if (length(slf_table) == 1) {
+    slf_table <- slf_table[[1]]
+  } else {
+    slf_table <- purrr::reduce(slf_table, dplyr::union_all)
   }
 
-  # Bind the files which have been read together
-  slf <- dplyr::bind_rows(slfs_list)
-
-  # With testing it is faster to remove any extra columns after binding
-  if (remove_partnership_var) {
-    slf <- dplyr::select(slf, -"hscp2018")
+  if (as_data_frame) {
+    return(dplyr::collect(slf_table))
+  } else {
+    return(slf_table)
   }
-  if (remove_recid_var) {
-    slf <- dplyr::select(slf, -"recid")
-  }
-
-  # Return the data as a tibble
-  slf <- tibble::as_tibble(slf, .name_repair = "minimal")
-
-  return(slf)
 }
 
 #' Read a Source Linkage episode file
 #'
-#' @inheritParams read_slf
-#' @inheritParams fst::read_fst
-#'
-#' @return a [tibble][tibble::tibble-package]
+#' @inherit read_slf
 #' @export
 #'
 #' @examples
 #' \dontrun{
 #' read_slf_episode("1718",
-#'   columns = c("anon_chi", "dob", "demographic_cohort"),
-#'   from = 100000,
-#'   to = 101000
+#'   col_select = c("anon_chi", "dob", "demographic_cohort")
 #' )
 #'
 #' read_slf_episode(c("1718", "1819"),
-#'   columns = c("anon_chi", "dob", "demographic_cohort"),
-#'   from = 100000,
-#'   to = 101000
+#'   col_select = c("anon_chi", "dob", "demographic_cohort"),
+#'   as_data_frame = FALSE
 #' )
 #' }
-read_slf_episode <-
-  function(year,
-           columns = NULL,
-           partnerships = NULL,
-           recids = NULL,
-           ...) {
-    # TODO add option to drop blank CHIs?
-    # TODO add a filter by recid option
-    return(
-      read_slf(
-        year = year,
-        file_version = "episode",
-        partnerships = unique(partnerships),
-        recids = unique(recids),
-        columns = unique(columns),
-        ...
-      )
+read_slf_episode <- function(
+    year,
+    col_select = NULL,
+    partnerships = NULL,
+    recids = NULL,
+    as_data_frame = TRUE,
+    dev = TRUE,
```

Comment:
```suggestion
    dev = FALSE,
```
Should dev be false here?